src/mcx_core.cu

```diff
@@ -1935,7 +1935,7 @@ __global__ void mcx_main_loop(uint media[], OutputType field[], float genergy[],
                 if (gcfg->outputtype == otEnergy) {
                     weight = w0 - p.w;
                 } else if (gcfg->outputtype == otFluence || gcfg->outputtype == otFlux) {
-                    weight = (prop.mua == 0.f) ? 0.f : ((w0 - p.w) / (prop.mua));
+                    weight = (prop.mua/prop.mus < 0.001) ? (w0 * len * (1 - prop.mua * len / 2)) : ((w0 - p.w) / (prop.mua));
```

Comment:
currently, mcx simulation kernel could accept domains with either mua or mus to be exactly 0. with your new formula
```
weight = (prop.mua/prop.mus < 0.001) ? (w0 * len * (1 - prop.mua * len / 2)) : ((w0 - p.w) / (prop.mua));
```

I see it can raise an exception if mus is 0, or both mua/mus is exactly 0.

please consider a formula that can also accommodate the case when mus=0.

Reply:
several minor comments:

1. for numerical constants, please add `f` at the end so that it won't be treated as "double constant"  which will trigger double-precision hardware/computation, which is slow on consumer GPUs.
2. same for integer constants, convert those to 1.f or 2.f could avoid unnecessary type conversions
3. avoid using division "/" in a CUDA code. it is one of the most expensive operators in the GPU kernel, even more expensive than `sin/cos`. try to convert to multiplication whenever possible, for example, instead of `/2` , change it to `*0.5f`